### PR TITLE
Get rid of IdentityManager call in ProfileSyncService

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves https://github.com/brave/brave-browser/issues/12984

The place in code where it happens is 
and here is a place in code where it happened
https://source.chromium.org/chromium/chromium/src/+/master:components/sync/driver/profile_sync_service.cc;l=962-970;drc=d81c5852498699fe3cd812e78d31c77c28e29281;bpv=1;bpt=1?q=ProfileSyncService::OnSyncCycleCompleted&ss=chromium%2Fchromium%2Fsrc

But we trying as much as possible to avoid patches. So in this PR we introduce own `BraveIdentityManagerFactory` which creates  `BraveIdentityManager` with stubbed method of `GetAccountsInCookieJar()` and `BravePrimaryAccountMutatorImpl` with stubbed method `ClearPrimaryAccount`

Fix of reverted  https://github.com/brave/brave-core/pull/7346 

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Open tab brave://net-export/
2. Start network logging with high available option (` Include raw bytes (will include cookies and credentials)`)
3. Enable sync, ensure you can see the device
4. Stop network logging
5. Analyze the log (suggested https://netlog-viewer.appspot.com/#import) 
6. Ensure there is no activity to `accounts.google.com`

### CODE REVIEW

File: patches/components-signin-internal-identity_manager-BUILD.gn.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
+diff --git a/components/signin/internal/identity_manager/BUILD.gn b/components/signin/internal/identity_manager/BUILD.gn
+index 893483064888ae2045114f0fd852993d3158b707..f294ffad85e99883315b4676966f3fec1e0a22d7 100644
+--- a/components/signin/internal/identity_manager/BUILD.gn
++++ b/components/signin/internal/identity_manager/BUILD.gn
+@@ -3,6 +3,7 @@
+ # found in the LICENSE file.
+ 
+ import("//components/signin/features.gni")
++import("//brave/components/signin/internal/identity_manager/sources.gni")
```
Comment (AlexeyBarabash) [inline patches/components-signin-internal-identity_manager-BUILD.gn.patch:9]: including through src/brave/build/config/brave_build.gni gives build errors for Chromium test targets like 
```
ERROR at //components/signin/internal/identity_manager/BUILD.gn:130:14: Undefined identifier
  sources += brave_components_signin_internal_identity_manager_sources
             ^--------------------------------------------------------
See //components/BUILD.gn:146:5: which caused the file to be included.
    "//components/signin/internal/identity_manager:unit_tests",
    ^---------------------------------------------------------

```

**Review by darkdh — APPROVED** [review]

**Comment by AlexeyBarabash** [comment]
iOS CI fails with unrelated to the PR error
```
21:42:04  Failing tests:
21:42:04  	TestTargetName:
21:42:04  		-[AdsWrapperTest testPreferencePersistance]
```

**Comment by AlexeyBarabash** [comment]
Command `npm run -- Debug` succeeded on macOS, Linux and Windows

**Comment by AlexeyBarabash** [comment]
Rebased branch

**Comment by AlexeyBarabash** [comment]
CI result:
iOS tets failed
```
17:00:45  Failing tests:
17:00:45  	TestTargetName:
17:00:45  		-[AdsWrapperTest testPreferencePersistance]
```
Not related to the PR https://ci.brave.com/job/pr-brave-browser-sync_no_identity_manager2-ios/4/execution/node/164/log/ 

failed lint https://ci.brave.com/job/pr-brave-browser-sync_no_identity_manager2-post-init/2/execution/node/136/log/, but local lint succeeded 
```
16:01:57  This change has GN build file diff for build/mac/BUILD.gn
16:01:57  This change has GN build file diff for build/mac/config.gni
16:01:57  
16:01:57  Format check failed. Run npm format to fix.
```
which is strange because the mentioned files were not touched by the PR; restarted that CI phase

**Comment by AlexeyBarabash** [comment]
re-run CI didn't help; rebased branch

**Comment by AlexeyBarabash** [comment]
failure of lint is known issue

File: chromium_src/components/sync/driver/profile_sync_service.h [hunk 2]
```diff
@@ -6,6 +6,8 @@
 #ifndef BRAVE_CHROMIUM_SRC_COMPONENTS_SYNC_DRIVER_PROFILE_SYNC_SERVICE_H_
 #define BRAVE_CHROMIUM_SRC_COMPONENTS_SYNC_DRIVER_PROFILE_SYNC_SERVICE_H_
 
+#include "brave/components/signin/public/identity_manager/brave_identity_manager.h"
```
Comment (mkarolin) [inline chromium_src/components/sync/driver/profile_sync_service.h:9]: Needed?

Comment (AlexeyBarabash) [reply chromium_src/components/sync/driver/profile_sync_service.h]: Yes, needed. But you are right, I have included at the wrong place.
Fixed at https://github.com/brave/brave-core/pull/7480/commits/6cc256d6f0a7e13a1887e85b7767de561bc5f2d6

**Review by mkarolin — APPROVED** [review]
`chromium_src` and `patches` changes LGTM

**Comment by AlexeyBarabash** [comment]
iOS build failed with
```

19:50:53  FAILED: gen/brave/vendor/brave-ios/ads.mojom.objc.h gen/brave/vendor/brave-ios/ads.mojom.objc+private.h gen/brave/vendor/brave-ios/ads.mojom.objc.mm gen/brave/vendor/brave-ios/objc_templates_bytecode/ads_objc_templates.zip 
19:50:53  python ../../brave/vendor/brave-ios/scripts/mojo/gen_model_wrappers.py --mojom-module=/Users/jenkins/jenkins/workspace/pr-brave-browser-sync_no_identity_manager2-ios/src/out/ios_Release_arm64/gen/brave/vendor/bat-native-ads/include/bat/ads/public/interfaces/ads.mojom-module --module-include-path=bat/ads/public/interfaces --mojom-file=/Users/jenkins/jenkins/workspace/pr-brave-browser-sync_no_identity_manager2-ios/src/brave/vendor/bat-native-ads/include/bat/ads/public/interfaces/ads.mojom --output-dir=/Users/jenkins/jenkins/workspace/pr-brave-browser-sync_no_identity_manager2-ios/src/out/ios_Release_arm64/gen/brave/vendor/brave-ios
19:50:53  Traceback (most recent call last):
19:50:53    File "../../brave/vendor/brave-ios/scripts/mojo/gen_model_wrappers.py", line 47, in <module>
19:50:53      sys.exit(main())
19:50:53    File "../../brave/vendor/brave-ios/scripts/mojo/gen_model_wrappers.py", line 44, in main
19:50:53      generator.GenerateFiles(output_dir)
19:50:53    File "/Users/jenkins/jenkins/workspace/pr-brave-browser-sync_no_identity_manager2-ios/src/brave/vendor/brave-ios/scripts/mojo/mojom_objc_generator.py", line 357, in GenerateFiles
19:50:53      self.Write(self._GenerateModuleHeader(),
19:50:53    File "/Users/jenkins/jenkins/workspace/pr-brave-browser-sync_no_identity_manager2-ios/src/brave/vendor/brave-ios/scripts/mojo/../../../../../mojo/public/tools/mojom/mojom/generate/template_expander.py", line 35, in GeneratorInternal
19:50:53      return ApplyTemplate(args[0], path_to_template, parameters, **kwargs)
19:50:53    File "/Users/jenkins/jenkins/workspace/pr-brave-browser-sync_no_identity_manager2-ios/src/brave/vendor/brave-ios/scripts/mojo/../../../../../mojo/public/tools/mojom/mojom/generate/template_expander.py", line 28, in ApplyTemplate
19:50:53      return template.render(params)
19:50:53    File "/Users/jenkins/jenkins/workspace/pr-brave-browser-sync_no_identity_manager2-ios/src/third_party/jinja2/environment.py", line 1008, in render
19:50:53      return self.environment.handle_exception(exc_info, True)
19:50:53    File "/Users/jenkins/jenkins/workspace/pr-brave-browser-sync_no_identity_manager2-ios/src/third_party/jinja2/environment.py", line 780, in handle_exception
19:50:53      reraise(exc_type, exc_value, tb)
19:50:53    File "/Users/jenkins/jenkins/workspace/pr-brave-browser-sync_no_identity_manager2-ios/src/out/ios_Release_arm64/gen/brave/vendor/brave-ios/objc_templates_bytecode/objc_templates.zip/tmpl_4b4c750a11f14c4ee990ca69a07408825b11d62d.pyc", line 29, in top-level template code
19:50:53  EOFError: EOF read where not expected
```
restarted


lint failed with the same weird error
```

19:50:25  Done processing vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/new_tab_page_ads/new_tab_page_ads_frequency_capping.cc
19:50:25  Running git cl/gn format on the diff from 48cf3ca1e25ee4ec10e1cd47a0d615d0652f384a
19:50:25  ...
19:50:29  This change has GN build file diff for build/mac/BUILD.gn
19:50:29  This change has GN build file diff for build/mac/config.gni
``` but succeeds locally

**Comment by AlexeyBarabash** [comment]
rebased to the latest b-c as it got Chromium update

**Comment by AlexeyBarabash** [comment]
CI succeeded except `audit-deps` for `smallvec` which is known issue. Merging.

**Pull request merged**